sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/CommunicationIdentityClientBuilder.java

```diff
@@ -203,6 +216,18 @@ private CommunicationIdentityClientImpl createServiceImpl() {
         return clientBuilder.buildClient();
     }
 
+    private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
+        if (this.tokenCredential != null) { 
+            return new BearerTokenAuthenticationPolicy(
+                this.tokenCredential, "https://communication.azure.com//.default");          
+        } else if (this.credential != null) {
+            return new HmacAuthenticationPolicy(this.credential);            
+        } else {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Missing credential information while building a client."));
+        }
```

Comment:
If both are set, do we want to throw or prioritize `tokenCredential`?

Reply:
If both are set, I think we should prioritize tokenCredential as it is now, right? It makes sense to use that one as our default.

Reply:
Talked offline, we're following `BlobClientBuilder`'s example now with a last one wins logic, and additionally info log that the previous credential got overridden.